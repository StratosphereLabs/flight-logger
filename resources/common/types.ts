import {
  TRPCErrorShape,
  TRPC_ERROR_CODE_KEY,
  TRPC_ERROR_CODE_NUMBER,
} from '@trpc/server/rpc';
import { AlertProps } from 'react-daisyui';
import { FieldValues, Path, UseControllerProps } from 'react-hook-form';
import { typeToFlattenedError } from 'zod';

export type GenericDataType = { id: string | number } & Record<string, unknown>;

export interface AlertMessage {
  status: AlertProps['status'];
  message: string;
}

export interface ErrorResponse {
  status: number;
  message: string;
}

export interface DefaultErrorData {
  [x: string]: unknown;
  code: TRPC_ERROR_CODE_KEY;
  httpStatus: number;
  path?: string;
  stack?: string;
  zodError?: typeToFlattenedError<Record<string, unknown>, string> | null;
}

export interface DefaultErrorShape
  extends TRPCErrorShape<TRPC_ERROR_CODE_NUMBER, DefaultErrorData> {
  message: string;
  code: TRPC_ERROR_CODE_NUMBER;
}

export interface FormFieldProps<Values extends FieldValues> {
  controllerProps?: Omit<UseControllerProps<Values>, 'name'>;
  isRequired?: boolean;
  labelText?: string;
  name: Path<Values>;
  placeholder?: string;
}

export interface PaginationMetadata {
  page: number;
  pageCount: number;
  limit: number;
  itemCount: number;
  pages: Array<number | null>;
}

export interface PaginatedResults<Data> {
  metadata: PaginationMetadata;
  results: Data[];
}

export interface Transform<TOutput> {
  output: (val: string) => TOutput;
  input: (val: TOutput) => string;
}
