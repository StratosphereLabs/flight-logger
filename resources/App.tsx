import { BrowserRouter, Route, Routes } from 'react-router-dom';
import { MainLayout } from './layouts';
import { AddFlight, Data, Flights, Home } from './pages';

export const App = (): JSX.Element => (
  <BrowserRouter>
    <Routes>
      <Route path="/" element={<MainLayout />}>
        <Route path="/" element={<Home />} />
        <Route path="add-flight" element={<AddFlight />} />
        <Route path="flights" element={<Flights />} />
        <Route path="data" element={<Data />} />
      </Route>
    </Routes>
  </BrowserRouter>
);

export default App;
