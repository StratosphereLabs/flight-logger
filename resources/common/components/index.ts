export * from './AlertMessages';
export * from './DarkModeButton';
export * from './DashboardContainer';
export * from './Form';
export * from './FormInput';
export * from './FullScreenLoader';
export * from './LoadingCard';
export * from './Table';
