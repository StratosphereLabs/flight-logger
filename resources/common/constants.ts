export const REST_API_URL = `${import.meta.env.VITE_APP_URL as string}/rest`;
export const TRPC_API_URL = `${import.meta.env.VITE_APP_URL as string}/trpc`;
export const GOOGLE_CLIENT_ID = import.meta.env.VITE_GOOGLE_CLIENT_ID as string;
