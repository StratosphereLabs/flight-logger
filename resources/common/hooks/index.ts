export * from './useAuthPage';
export * from './useDebouncedValue';
export * from './useErrorResponseHandler';
export * from './useFocusOnFirstField';
export * from './useGoogleLoginMutation';
export * from './useOutsideClick';
export * from './useProtectedPage';
export * from './useScrollBar';
export * from './useSuccessResponseHandler';
export * from './useTRPCErrorHandler';
export * from './useTypeaheadInput';
