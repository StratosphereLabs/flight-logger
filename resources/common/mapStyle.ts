export const darkModeStyle = [
  { elementType: 'geometry', stylers: [{ color: '#242f3e' }] },
  { elementType: 'labels.text.stroke', stylers: [{ color: '#242f3e' }] },
  { elementType: 'labels.text.fill', stylers: [{ color: '#746855' }] },
  {
    featureType: 'administrative.locality',
    elementType: 'all',
    stylers: [{ visibility: 'off' }],
  },
  {
    featureType: 'poi',
    elementType: 'all',
    stylers: [{ visibility: 'off' }],
  },
  {
    featureType: 'road',
    elementType: 'all',
    stylers: [{ visibility: 'off' }],
  },
  {
    featureType: 'transit',
    elementType: 'all',
    stylers: [{ visibility: 'off' }],
  },
  {
    featureType: 'water',
    elementType: 'geometry',
    stylers: [{ color: '#17263c' }],
  },
  {
    featureType: 'water',
    elementType: 'labels.text.fill',
    stylers: [{ color: '#515c6d' }],
  },
  {
    featureType: 'water',
    elementType: 'labels.text.stroke',
    stylers: [{ color: '#17263c' }],
  },
];

export const lightModeStyle = [
  {
    featureType: 'administrative.locality',
    elementType: 'all',
    stylers: [{ visibility: 'off' }],
  },
  {
    featureType: 'poi',
    elementType: 'all',
    stylers: [{ visibility: 'off' }],
  },
  {
    featureType: 'road',
    elementType: 'all',
    stylers: [{ visibility: 'off' }],
  },
  {
    featureType: 'transit',
    elementType: 'all',
    stylers: [{ visibility: 'off' }],
  },
];
