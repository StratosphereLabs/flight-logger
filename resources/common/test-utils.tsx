import {
  render,
  type RenderOptions,
  type RenderResult,
} from '@testing-library/react';
import { type InitialEntry } from 'history';
import { type ReactElement } from 'react';
import { AppWrapper } from '../AppWrapper';

interface CustomRenderOptions extends Omit<RenderOptions, 'wrapper'> {
  initialEntries?: InitialEntry[];
}

const customRender = (
  ui: ReactElement,
  options?: CustomRenderOptions,
): RenderResult =>
  render(ui, {
    wrapper: AppWrapper,
    ...options,
  });

/* eslint-disable import/export */
export * from '@testing-library/react';

export { customRender as render };
