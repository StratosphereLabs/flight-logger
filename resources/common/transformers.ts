import { Transform } from 'stratosphere-ui';

export const numberInputTransformer: Transform<number | null> = {
  input: value => (value !== null ? value.toString() : ''),
  output: value => (value.length > 0 ? parseInt(value) : null),
};

export const nullEmptyStringTransformer: Transform<string | null> = {
  input: value => (value !== null ? value : ''),
  output: value => (value.length > 0 ? value : null),
};
