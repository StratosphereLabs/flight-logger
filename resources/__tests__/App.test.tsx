import { expect, it } from 'vitest';
import App from '../App';
import { render } from '../common/test-utils';

it('renders app', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
