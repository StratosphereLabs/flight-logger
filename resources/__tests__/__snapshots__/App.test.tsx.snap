// Vitest Snapshot v1

exports[`renders app 1`] = `
<div>
  <div
    class="pb-40 flex w-full component-preview p-4 items-center justify-center gap-2 font-sans"
  >
    <div
      aria-label="Navbar"
      class="navbar bg-base-100 shadow-xl rounded-box"
      role="navigation"
    >
      <div
        class="flex-1 navbar-start"
      >
        <div
          class="dropdown"
          role="listbox"
        >
          <label
            tabindex="0"
          >
            <button
              class="btn lg:hidden btn-ghost"
              tabindex="0"
            >
              <svg
                class="h-5 w-5"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4 6h16M4 12h8m-8 6h16"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
            <ul
              class="dropdown-content menu p-2 shadow bg-base-100 rounded-box w-52 menu-compact mt-3"
              tabindex="0"
            >
              <a
                href="/"
              >
                <li>
                  <a>
                    Home
                  </a>
                </li>
              </a>
              <a
                href="/flights"
              >
                <li>
                  <a>
                    My Flights
                  </a>
                </li>
              </a>
              <a
                href="/data"
              >
                <li>
                  <a>
                    Data
                  </a>
                </li>
              </a>
            </ul>
          </label>
          <ul
            class="dropdown-content"
          />
        </div>
        <a
          class="btn btn-ghost normal-case text-xl"
          href="/"
        >
          FlightLogger
        </a>
      </div>
      <div
        class="hidden lg:flex navbar-center"
      >
        <ul
          class="menu p-0 menu-horizontal"
          role="menu"
        >
          <li
            class=""
            role="menuitem"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class=""
            role="menuitem"
          >
            <a
              class=""
              href="/flights"
            >
              My Flights
            </a>
          </li>
          <li
            class=""
            role="menuitem"
          >
            <a
              class=""
              href="/data"
            >
              Data
            </a>
          </li>
        </ul>
      </div>
      <div
        class="flex-1 navbar-end"
      >
        <a
          class="btn"
          href="/add-flight"
        >
          Add Flight
        </a>
      </div>
    </div>
  </div>
</div>
`;
